Description: Stop using apt-key to setup secure apt, and instead copy the gpg key snippet.
Author: Dimitri John Ledkov <xnox@ubuntu.com>


Index: live-build-3.0~a57/scripts/build/lb_chroot_archives
===================================================================
--- live-build-3.0~a57.orig/scripts/build/lb_chroot_archives
+++ live-build-3.0~a57/scripts/build/lb_chroot_archives
@@ -249,10 +249,10 @@ EOF
 					# Adding archive signing keys (chroot)
 					if [ -e "${_BASE}/archives/${REPOSITORY}.key.chroot" ]
 					then
-						cat "${_BASE}/archives/${REPOSITORY}.key.chroot" | Chroot chroot "apt-key add -"
+					    cp "${_BASE}/archives/${REPOSITORY}.key.chroot" chroot/etc/apt/trusted.gpg.d/${REPOSITORY}.gpg
 					elif [ -e "${_BASE}/archives/${REPOSITORY}.key" ]
 					then
-						cat "${_BASE}/archives/${REPOSITORY}.key" | Chroot chroot "apt-key add -"
+					    cp "${_BASE}/archives/${REPOSITORY}.key" chroot/etc/apt/trusted.gpg.d/${REPOSITORY}.gpg
 					fi
 				fi
 			done
@@ -516,9 +516,7 @@ EOF
 				do
 					if [ -e "${FILE}" ]
 					then
-						cp ${FILE} chroot/root
-						Chroot chroot "apt-key add /root/$(basename ${FILE})"
-						rm -f chroot/root/$(basename ${FILE})
+						cp ${FILE} chroot/etc/apt/trusted.gpg.d/$(basename ${FILE}).gpg
 					fi
 				done
 			fi
@@ -804,10 +802,10 @@ EOF
 					# Adding archive signing keys (binary)
 					if [ -e "${_BASE}/archives/${REPOSITORY}.key.binary" ]
 					then
-						cat "${_BASE}/archives/${REPOSITORY}.key.binary" | Chroot chroot "apt-key add -"
+						cp "${_BASE}/archives/${REPOSITORY}.key.binary" chroot/etc/apt/trusted.gpg.d/${REPOSITORY}.gpg
 					elif [ -e "${_BASE}/archives/${REPOSITORY}.key" ]
 					then
-						cat "${_BASE}/archives/${REPOSITORY}.key" | Chroot chroot "apt-key add -"
+						cat "${_BASE}/archives/${REPOSITORY}.key" chroot/etc/apt/trusted.gpg.d/${REPOSITORY}.gpg
 					fi
 				fi
 			done
